Firmware/ControladorRobo/ControladorRobo.ino

```diff
@@ -81,23 +80,23 @@ ISR(TIMER1_COMPA_vect)
 {
     // Calculo das velocidades dos motores
 
-    float arcRight = ((float(pulsosDir))/3200.0)*M_PI;
-    float arcLeft = ((float(pulsosEsq))/3200.0)*M_PI;
+    float rightAngularDelta = ((float)right.read()/ENCODER_RESOLUTION)*M_PI;
+    float leftAngularDelta = ((float)left.read()/ENCODER_RESOLUTION)*M_PI;
 
-    wEsq = arcLeft*freq;
-    wDir = arcRight*freq;
+    float rightAngularSpeed = rightAngularDelta*freq;
+    float leftAngularSpeed = leftAngularDelta*freq;
 
-    posDir+=arcRight;
-    posEsq+=arcLeft;
+    rightAngPos += rightAngularDelta;
+    leftAngPos += leftAngularDelta;
 
-    pulsosDir = 0;
-    pulsosEsq = 0;
+    left.write(0);
+    right.write(0);
 
-    float u_esq = controle_esq.update(refEsq, wEsq);
-    float u_dir = controle_dir.update(refDir, wDir);
+    float u_esq = controle_esq.update(refEsq, leftAngularSpeed);
+    float u_dir = controle_dir.update(refDir, rightAngularSpeed);
```

Comment:
*AngularSpeed doesn't make sense to be long if the value is converted to float in Controller::update